test-programs/compressed-token-test/tests/test.rs

```diff
@@ -237,27 +242,418 @@ async fn test_mint_to<const MINTS: usize, const ITER: usize>() {
+            mint: mint_1.clone(),
+            token_pool_pda: mint_pool_1,
+            token_program: anchor_spl::token::ID,
+            light_system_program: light_system_program::ID,
+            registered_program_pda: light_system_program::utils::get_registered_program_pda(
+                &light_system_program::ID,
+            ),
+            noop_program: Pubkey::new_from_array(
+                account_compression::utils::constants::NOOP_PUBKEY,
+            ),
+            account_compression_authority: invalid_account_compression_authority.pubkey(),
+            account_compression_program: account_compression::ID,
+            merkle_tree: merkle_tree_pubkey,
+            self_program: light_compressed_token::ID,
+            system_program: system_program::ID,
+        };
+        let instruction = Instruction {
+            program_id: light_compressed_token::ID,
+            accounts: accounts.to_account_metas(Some(true)),
+            data: instruction_data.data(),
+        };
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            anchor_lang::error::ErrorCode::ConstraintSeeds.into(),
+        )
+        .unwrap();
+    }
+    // 9. Invalid Merkle tree.
+    {
+        let invalid_merkle_tree = Keypair::new();
+        let instruction = create_mint_to_instruction(
+            &payer_1.pubkey(),
+            &payer_1.pubkey(),
+            &mint_1,
+            &invalid_merkle_tree.pubkey(),
+            amounts.clone(),
+            recipients.clone(),
+        );
+        rpc.create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await
+            .unwrap_err(); // Anchor panics, there is no specific error to assert.
+    }
+    // 10. Mint more than `u64::MAX` tokens.
+    {
+        // Overall sum greater than `u64::MAX`
+        let amounts = vec![u64::MAX / 5; MINTS];
+        let instruction = create_mint_to_instruction(
+            &payer_1.pubkey(),
+            &payer_1.pubkey(),
+            &mint_1,
+            &merkle_tree_pubkey,
+            amounts,
+            recipients.clone(),
+        );
+        let result = rpc
+            .create_and_send_transaction(&[instruction], &payer_1.pubkey(), &[&payer_1])
+            .await;
+        assert_rpc_error(
+            result,
+            0,
+            light_compressed_token::ErrorCode::MintTooLarge.into(),
+        )
+        .unwrap();
+    }
+    // 11. Multiple mints which overflow the token supply over `u64::MAX`.
+    {
+        let amounts = vec![u64::MAX / 10; MINTS];
+        let instruction = create_mint_to_instruction(
+            &payer_1.pubkey(),
+            &payer_1.pubkey(),
+            &mint_1,
+            &merkle_tree_pubkey,
+            amounts,
+            recipients.clone(),
+        );
+        // The first mint is still below `u64::MAX`.
```

Comment:
do we have an explicit test for mint zero?

Reply:
Sort of, here https://github.com/Lightprotocol/light-protocol/blob/4f037c37fbc65af096c04c18ffac649e41f696fc/test-programs/compressed-token-test/tests/test.rs#L252

`0` is one of the inputs, among other `MAX` values, but I think it's good enough for checking.

But let me add a test just with zeros.

Reply:
Done in https://github.com/Lightprotocol/light-protocol/pull/964/commits/38334809a8b3897e8f9f6a62ade0994de493fcf0